pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -317,3 +308,22 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 	return result, false, nil
 
 }
+
+func validateASAPDependencies(context *wiringplugin.WiringContext) error {
+	asapDependencyCount := 0
+	for _, dep := range context.Dependencies {
+		_, found, err := knownshapes.FindASAPKeyShapes(dep.Contract.Shapes)
+		if err != nil {
+			return err
```

Comment:
If this becomes the error on the State object in Kubernetes, do we want to wrap this with a useful message for the user?

At the moment it will read 
```
... errors.Wrapf(err, "failed to copy shape %T into %T", shape, copyInto)
```

Reply:
You phrased this as a question, I am not sure if you are just being polite or if you are making a suggestion. Either way I would defer to the teams expertise here.